frappe/templates/includes/comments/comments.py

```diff
@@ -3,29 +3,45 @@
 from __future__ import unicode_literals
 
 import frappe
-import frappe.utils
+import re
 from frappe.website.render import clear_cache
+from frappe.utils import add_to_date, now
 
 from frappe import _
 
-@frappe.whitelist(allow_guest=True)
+@frappe.whitelist()
 def add_comment(comment, comment_email, comment_by, reference_doctype, reference_name, route):
 	doc = frappe.get_doc(reference_doctype, reference_name)
 
-	if len(comment) < 10:
-		frappe.msgprint(_('Comment Should be atleast 10 characters'))
-		return ''
+	if not comment.strip() or len(comment) < 10:
+		frappe.msgprint(_('Your comment should be atleast 10 characters long'))
+		return False
 
-	blacklist = ['http://', 'https://', '@gmail.com']
+	url_regex = re.compile(r"http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+", re.IGNORECASE)
+	email_regex = re.compile(r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)", re.IGNORECASE)
 
-	if any([b in comment for b in blacklist]):
+	if url_regex.match(comment) or email_regex.match(comment):
 		frappe.msgprint(_('Comments cannot have links or email addresses'))
-		return ''
+		return False
+
+	if not comment_email == frappe.session.user:
+		frappe.msgprint(_('Please comment as the logged in user'))
```

Comment:
This message is confusing... maybe something simpler?

or implicitly set `frappe.session.user` as email without raising.